components/visuals/TravelsOfDepartmentPrismMap.tsx

```diff
@@ -0,0 +1,136 @@
+import { Department } from "@prisma/client";
+import {
+  AccumulativeShadows,
+  Environment,
+  OrbitControls,
+} from "@react-three/drei";
+import { Canvas } from "@react-three/fiber";
+import { scaleLinear, scaleQuantile, schemeBlues } from "d3";
+import { GeoProjection } from "d3-geo";
+import { GeoJsonProperties } from "geojson";
+import { FC, useCallback, useMemo, useState } from "react";
+import { ExtrudeGeometryOptions } from "three";
+import type { Topology } from "topojson-specification";
+import { BtorsGroupedByCountryByDepartment } from "../../lib/data/queries/btors/getBtorsGroupedByCountryByDepartment";
+import { FeatureIdentifier } from "../../types/FeatureIdentifier";
+import KPI from "../KPI";
+import { MemoizedPrismMap } from "../PrismMap";
+import SoftLight from "../SoftLight";
+import Tooltip from "../Tooltip";
+
+type Props = {
+  btorsByCountryByDepartment: BtorsGroupedByCountryByDepartment;
+  topology: Topology;
+  topologyObject: string;
+  projection: GeoProjection;
+  width: number;
+  length: number;
+  extrudeGeometryOptions?: ExtrudeGeometryOptions;
+  department: Department;
+};
+
+const TravelsOfDepartmentPrismMap: FC<Props> = ({
+  btorsByCountryByDepartment,
+  topology,
+  topologyObject,
+  projection,
+  width,
+  length,
+  extrudeGeometryOptions = {},
+  department,
+}) => {
+  const [hoverInfo, setHoverInfo] = useState<FeatureIdentifier | undefined>(
+    undefined,
+  );
+  const featureProperties = useMemo(
+    () =>
+      new Map(
+        btorsByCountryByDepartment
+          .get(department.id)
+          ?.map(({ country, count }) => [country, { count }]),
+      ),
+    [department, btorsByCountryByDepartment],
+  );
+  const colorScale = useMemo(
+    () =>
+      scaleQuantile<string, string>()
+        .domain(Array.from(featureProperties.values()).map((d) => d.count))
+        .range(schemeBlues[5])
+        .unknown("ligthgrey"),
+    [featureProperties],
+  );
+
+  const propertyAccessor = useCallback(
+    (properties: GeoJsonProperties) => properties?.count,
+    [],
+  );
+  const onPointerEnterHandler = useCallback(
+    (d: FeatureIdentifier) => setHoverInfo(d),
+    [],
+  );
+  const onPointerLeaveHandler = useCallback(() => setHoverInfo(undefined), []);
+  const extrusionScale = useMemo(
+    () => scaleLinear().domain([0, 100]).range([0.01, 2]).unknown(0.01),
+    [],
+  );
+  return (
+    <div>
+      <Tooltip.Root open={!!hoverInfo} followCursor placement="top-start">
+        <Tooltip.Trigger asChild>
+          <div className="h-[400px] w-[600px]">
```

Comment:
Setting the `div`s withd to an absolute pixel value (for all breakpoints), does result in a design which is not really working for small screens from my point of view. What's your reasoning behind that @Pei-Nie? I created a `<CanvasStage />` which sets a default height (can be overwritten with `tailwind` classNames). Do you think this could be useful here? If not let me know how we could perhaps improve this component. Thank you!

Reply:
Ok.

Reply:
Jakob, for this issue, i replace the <div> with <CanvasStage> and assign classname property to <CanvasStage> like <CanvasStage className="h-[500px] w-[600px]">, if don't overwritten with className, the PrismMap will occupy the whole screen from left to right, which is not match with others maps.

Reply:
OK, I believe I was not precise enough here: actually it doesn't matter that much where you use the `className` `w-[600px]`. That problem is rather using it at all. Because we're aiming for a responsive layout, using widths defined in pixel are likely to cause trouble. I believe using `max-width` is what we want. E.g., like we did here: https://github.com/jakoblistabarth/itc-atlas/blob/3ebcf679ba5ddc9686b4dec1dd8e66debb1d3ba5/pages/introduction/travels.tsx#L115

Reply:
Ok, i see. I set default height (500px, same as Phd these map) of CanvasStage in Travel‘s prismMap component and use max-width to set the width of this map in the page.